docs/enhancements/0002_day0-integration.md

```diff
@@ -0,0 +1,285 @@
+# Day-0 integration with KMM
+
+### Motivation
+
+KMM is managing mostly day2 kmods in OCP. It also offers a library for generating
+systemd files for helping customers to deploy their day-1 kmod.
+
+What KMM doesn't offers today to customers is the deployment and management of
+day-0 kmods.
+
+Day-0 kmod are complex as they need to be present in the system at boot time,
+even before OCP is fully installed and certaily before KMM is deployed to the
+cluster, therefore, there is no much KMM can do to help deploy those kmods but
+it can manage their lifecycle once a customer has installed a cluster with his
+day-0 kmods in it - this is what this enhancement is about.
+
+### Day-0 cluster background
+
+A customer wishing to deploy OCP with custom nodes will need to generate 2
+artifacts using [image-builder](https://www.redhat.com/en/topics/linux/what-is-an-image-builder)
+  * A custom ISO for the nodes
+  * A container image encapsulating the ostree-commit that is deployed in the ISO
+
+The ISO will be used by [assisted-installer](https://docs.openshift.com/container-platform/4.14/installing/installing_on_prem_assisted/installing-on-prem-assisted.html)
+to install the cluster and the container image will be used to notify
+[MCO](https://docs.openshift.com/container-platform/4.14/post_installation_configuration/machine-configuration-tasks.html#understanding-the-machine-config-operator)
+that the image on the nodes is a custom image.
+
+When a customers has installed a "day-0 cluster" as mentioned above, we can
+assume that there is a `MachineConfig` in the cluster that looks something like
+```
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  creationTimestamp: "2023-12-31T14:03:08Z"
+  generation: 1
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 99-ybettan-external-image
+  resourceVersion: "1758"
+  uid: d1f07169-4cb5-4b85-93b3-3cd80530f028
+spec:
+  osImageURL: quay.io/ybettan/rhcos:414.92.202312311229-0
+```
+The important piece in it is `osImageURL: quay.io/ybettan/rhcos:414.92.202312311229-0`
+which tells MCO what that the ostree-commit that is supposed to be running on
+the node is encapsulated in `quay.io/ybettan/rhcos:414.92.202312311229-0`.
+
+If MCO, finds out that the OS on the node is different than the image specified
+in `osImageURL` then MCO will override the node OS based on the `MachineConfig`
+and reboot - we are going to leverage this mechanisem for the integration with
+KMM.
```

Comment:
```suggestion
If MCO finds out that the OS on the node is different than the image specified
in `osImageURL` then it will override the node OS based on the `MachineConfig`
and reboot - we are going to leverage this mechanism for the integration with
KMM.
```